Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp

```diff
@@ -559,8 +559,7 @@ CSSPixels BlockFormattingContext::compute_auto_height_for_block_level_element(Bo
 
             auto const& child_box_state = m_state.get(*child_box);
 
-            // Ignore anonymous block containers with no lines. These don't count as in-flow block boxes.
-            if (!child_box->is_table_wrapper() && child_box->is_anonymous() && child_box->is_block_container() && child_box_state.line_boxes.is_empty())
+            if (margins_collapse_through(*child_box, m_state))
```

Comment:
unfortunately `margins_collapse_through()` is a hack and currently returns true for everything with zero height instead of doing what the spec says. any chance you can find some way around to fix the issue without using it?

Reply:
now when I am actually looking into how would I solve this issue it seems like usage of `margins_collapse_through` is ok here. we will have to make `margins_collapse_through` follow the spec anyway but it is another issue :)